src/components/transaction/GetTransactionsId.tsx

```diff
@@ -23,45 +30,39 @@ const GetTransactionsId = () => {
           value={accountId}
           placeholder="ccc3a91d-449c-41ff-a6fe-d79001431e4f"
           onChange={(e) => setAccountId(e.target.value)}
-        ></input>
+        />
         <button
           onClick={() => {
             const validateInput = validateAccountId(accountId)
             if (validateInput) {
               setErrorMessage(validateInput)
-              setRequestStatus('error')
+
+              //to prevent fetching/refetching with invalid uuid
+              setSubmittedAccountId('')
             } else {
-              setRequestStatus('fetching')
-              api
-                .get(`accounts/${accountId}/transactions`)
-                .json<Array<TransactionResponse>>()
-                .then((data) => {
-                  setTransactionsForId(data)
-                  setRequestStatus('success')
-                })
-                .catch((e) => {
-                  setRequestStatus('error')
-                  if (e instanceof HTTPError) {
-                    e.response.json().then((e) => setErrorMessage(e.message))
-                  }
-                  setErrorMessage(e.message)
-                })
+              setSubmittedAccountId(accountId)
+              //must refetch since transaction won't show up if it is searched for,
+              //created, then searched for again; stays stuck on
+              //"No transactions found" unless manually refetched
```

Comment:
ahh so this is actually where using React-Query absolutely shines (or sucks depending on how its used hehe). This reminded me that mutations should _always_ invalidate caches that they will change. 

For instance the 3 mutations that create a transaction should invalidate both the query to get all transactions and the query to get transactions by Id but only the one matching the same accountId.